### DISCUSSION TITLE
dotenv-expand' "default" expansion not working

### DISCUSSION THREAD

**Discussion by FezVrasta** [vercel/next.js#34766] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 3.2.0
      pnpm: N/A
    Relevant packages:
      next: 12.1.0
      react: 17.0.2
      react-dom: 17.0.2

warn  - Latest canary version not detected, detected: "12.1.0", newest: "12.1.1-canary.1".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
      

### Describe the Bug

If I add a fallback/default value to one of my dotenv variable expansions the operation doesn't succeed.

```
NEXT_PUBLIC_TEST=${KEY:-default}
```

Results in:

```
> console.log(process.env.NEXT_PUBLIC_TEST);
< ":-default}"
```

### Expected Behavior

```
> console.log(process.env.NEXT_PUBLIC_TEST);
< "default"
```

This is explained in the dotenv-expand documentation here:
https://github.com/motdotla/dotenv-expand#what-rules-does-the-expansion-engine-follow

### To Reproduce

Add the above config to `.env` and run `yarn build`.

I noticed that the compiled source of @next/env is missing several parts of the dotenv-expand package (that is for some reason bundled in rather than referenced as external)

**Comment by balazsorban44** [comment]
This is because `@next/env` is locked on `dotenv-expand@5`, which does not have this feature. I opened a PR to bump it to the latest version, but there is a chance this would be a breaking change in Next.js, so please follow the PR at #34765

**Reply by FezVrasta** [reply]
If bumping is not an option could `@next/env` not bundle/inline `dotenv-expand` so that its version can be configured using Yarn resolutions maybe? (or patched through `yarn patch`)
